nrf52: nrf52840_dk fails to build — Adafruit BSP cores cannot find variant.h

## Symptom

[CI workflow `nrf52840_dk`](https://github.com/FastLED/FastLED/actions/workflows/build_nrf52840_dk.yml) fails on master with:

```
.../cores/nRF5/Uart.h:27:10: fatal error: variant.h: No such file or directory
.../cores/nRF5/delay.h:28:10: fatal error: variant.h: No such file or directory
```

The Adafruit BSP cores `#include "variant.h"`, which lives in `variants/<variant>/variant.h`. fbuild apparently isn't adding the variant directory to the include path when compiling the BSP cores for this board.

## Diagnosis context

- `supermini_nrf52840` (also Adafruit BSP) is **not** affected by this error — it hits a separate FastLED-side bug fixed in #2627.
- The variant alias `nRF52DK → pca10056` was added in [FastLED/fbuild#322](https://github.com/FastLED/fbuild/pull/322) (shipped in fbuild 2.2.11).
- The include-path miss is therefore likely in fbuild, not in the FastLED source.

## Acceptance

- Fix lives in fbuild repo (clone, fix, mint new version, ingest into `pyproject.toml`).
- This issue auto-closes when the FastLED PR that ingests the fix merges.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,17 +5,19 @@ description = "Add your description here"
 readme = "README.md"
 requires-python = ">=3.11"
 dependencies = [
-    # Pin to fbuild 2.2.11 — content of 2.2.10 republished with the full
-    # wheel set (win_amd64 + manylinux_x86_64 were missing from 2.2.10's
-    # PyPI upload). Bundles SAMD51 `cores/arduino` fallback + include-path
-    # injection (FastLED/fbuild#319/#320), nRF52 nRF52DK->pca10056 variant
-    # alias (FastLED/fbuild#321/#322), board JSON tinyuf2 partition rename
-    # + cmsis_dsp_lib validator whitelist (FastLED/fbuild#318), and
-    # daemon.rs subprocess-spawn lint annotations (FastLED/fbuild#317).
-    # Together these unblock Build SAMD51J, Build nRF52840 DK, and reduce
-    # Validate Boards drift from 39 boards to 14 (configuration-choice
-    # remainder pending maintainer review).
-    "fbuild==2.2.11",
+    # Pin to fbuild 2.2.12 — adds the nRF52 variant include-path alias
+    # (FastLED/fbuild#325/#326), MCU-aware linker-script alias for the
+    # PIO-named nrf52_xxaa.ld with globbed highest-version SoftDevice
+    # script (FastLED/fbuild#327/#328 — also fingerprints the resolved
+    # linker basename into Nrf52FingerprintMetadata so fast-path keys
+    # invalidate when the alias moves between actual files), and a
+    # release-auto CI fix bypassing soldr's corrupted cargo-zigbuild
+    # bin cache (FastLED/fbuild#331/#332). Together these unblock
+    # Build nRF52840 DK and Build nRF52 XIAO BLE Sense workflows by
+    # making fbuild find `variants/pca10056/` and
+    # `variants/Seeed_XIAO_nRF52840_Sense/` for board JSONs that track
+    # PIO/sandeepmistry upstream naming.
+    "fbuild==2.2.12",
     "platformio>=6.1.19,<6.2",
     "python-dateutil",
     "ruff",

